src/ontology/mondo-ingest.Makefile

```diff
@@ -516,28 +516,38 @@ slurp-all: $(foreach n,$(ALL_COMPONENT_IDS), slurp-$(n))
 sync: sync-subclassof
 
 .PHONY: sync-subclassof
-sync-subclassof: $(REPORTDIR)/sync-subClassOf.direct-in-mondo-only.tsv
+sync-subclassof: $(foreach n,$(ALL_COMPONENT_IDS), $(REPORTDIR)/$(n).subclass.confirmed.robot.tsv)
+	@rm -f $(REPORTDIR)/sync-subClassOf.confirmed.tsv
+	@rm -f $(REPORTDIR)/sync-subClassOf.direct-in-mondo-only.tsv
+	@rm -f $(TMPDIR)/sync-subClassOf.added.self_parentage_issues.tsv
+	$(MAKE) $(REPORTDIR)/sync-subClassOf.confirmed.tsv
+	$(MAKE) $(REPORTDIR)/sync-subClassOf.direct-in-mondo-only.tsv
+	$(MAKE) $(TMPDIR)/sync-subClassOf.added.self-parentage-issues.tsv
 
 # todo: drop this? This is really just an alias here for quality of life, but not used by anything.
 .PHONY: sync-subclassof-%
-sync-subclassof-%: $(REPORTDIR)/%.subclass.direct-in-mondo-only.tsv
-	@echo "$@ completed"
+sync-subclassof-%: $(REPORTDIR)/%.subclass.confirmed.robot.tsv
```

Comment:
## Other `sync-subclassof`-related `makefile` changes
While adding the goal for `$(TMPDIR)/sync-subClassOf.added.self-parentage.tsv`, the way I had things set up made less sense, and I noticed some things w/ how the dependency structure, etc, were set up that was just generally confusing. I updated several of these goals in several ways  to make the setup easier to understand and maintain

I also added some minor fixes / optimizations. For example, some `rm -f` statements that fix edge cases that can happen if you run / re-run parts of this pipeline locally, as opposed to a fresh build. I also changed one of the goal prereqs from a `.PHONY` alias goal to the actual goal represented by that alias, which in certain situations should prevent this goal from running unnecessarily. This also comes up only in local development / non-build context.

Reply:
Folding this `rm` pattern into the make pipelines itself should be done sparingly as it undermines the whole point of using a makefile: being able to avoid rebuilding files if none of the dependencies have changed. 

using `rm` pattern makes sense in development, not in production IMO, but just take it under advisement, not as a blocker for this PR.

Reply:
I thought about this and responded / made a change. See: [comment](https://github.com/monarch-initiative/mondo-ingest/pull/501#discussion_r1622929145)